src/azul/openapi/validation.py

```diff
@@ -0,0 +1,147 @@
+import json
+import logging
+from typing import (
+    Mapping,
+)
+
+from chalice import (
+    BadRequestError,
+)
+import jsonschema
+from more_itertools import (
+    one,
+)
+
+from azul.openapi.schema import (
+    type_lookup,
+)
+from azul.uuids import (
+    InvalidUUIDError,
+    InvalidUUIDVersionError,
+    validate_uuid,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SpecValidator:
+
+    def __init__(self,
+                 content: Mapping = None,
+                 pattern: str = None,
+                 name: str = None,
+                 required: bool = None,
+                 schema: Mapping = None,
+                 **kwargs):
+        """
+        :param kwargs: Used to extract the `in` parameter.
```

Comment:
unexpected kwargs would cause failure in the integration tests. due to a mismatch in the Open-API specification. 

https://github.com/DataBiosphere/azul/blob/32b462d2f32d3de3f17b3a3a2128515bb8a3a6d6/test/integration_test.py#L568

I do think we can perform the validation at the unit-test level rather than waiting for the integration-test to be run // infra to be deployed.. Because the spec is the same as the runtime, we don't have to inspect for unexpected kwargs.

Reply:
added openapi spec validation test to ServiceSpecValidation, removed from IT